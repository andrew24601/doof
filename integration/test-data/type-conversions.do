// Test type conversions
// Tests: INT_TO_FLOAT, FLOAT_TO_INT, INT_TO_DOUBLE, DOUBLE_TO_INT, FLOAT_TO_DOUBLE, DOUBLE_TO_FLOAT

function main(): int {
    let i: int = 42;
    let f: float = 3.14;
    let d: double = 2.71828;
    
    let i_to_f: float = i;          // INT_TO_FLOAT (implicit)
    let f_to_i: int = f;            // FLOAT_TO_INT (implicit, truncates)
    let i_to_d: double = i;         // INT_TO_DOUBLE (implicit)
    let d_to_i: int = d;            // DOUBLE_TO_INT (implicit, truncates)
    let f_to_d: double = f;         // FLOAT_TO_DOUBLE (implicit)
    let d_to_f: float = d;          // DOUBLE_TO_FLOAT (implicit)
    
    println(i_to_f);    // Should print 42
    println(f_to_i);    // Should print 3
    println(i_to_d);    // Should print 42
    println(d_to_i);    // Should print 2
    println(f_to_d);    // Should print 3.14
    println(d_to_f);    // Should print 2.71828
    
    return 0;
}
