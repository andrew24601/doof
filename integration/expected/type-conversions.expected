42
3
42
2
3.14
2.71828
